src/model_extractors.py

```diff
@@ -0,0 +1,95 @@
+import random
+from dataclasses import dataclass
+from typing import Dict, List
+
+import torch
+from simpletransformers.seq2seq import Seq2SeqArgs, Seq2SeqModel
+from torch.utils.data import DataLoader
+from transformers import AutoModel
+
+from src.data import QuestionAnsweringInputEncoder, QuestionAnsweringModelInput
+from src.models.qa import QuestionAnsweringModel
+from src.stop_watch import time_block
+
+
+@dataclass
+class LabeledSegment:
+    segment: str
+    label: str
+    score: float
+
+    def to_dict(self):
+        return self.__dict__
+
+
```

Comment:
Consider creating a `Seq2seqPredictor` that follows the same interface as the [`BenefitsAndDrawbacksExtractor`](https://github.com/choosistant/algorithms/blob/c6ae8c1b0d54f3588ba9642c42aa0915eb31aa13/src/model_extractors.py#L25):

```python
class Predictor(Protocol):
    """An interface for a model that can predict labels for segments of text."""
    def predict(self, document: str) -> List[LabeledSegment]:
        ...
```
Here is an example:
```python
class Seq2SecPredictor:
    """A seq2seq model that predicts labels for segments of text."""
    def __init__(
        self,
        model_name: str,
        cuda_device_no: int,
        encoder_decoder_type: str="bart",
        encoder_decoder_name: str="facebook/bart-large",
    ) -> None:
        self._backbone_model = AutoModel.from_pretrained(model_name)
        model_args = Seq2SeqArgs()
        self._model = Seq2SeqModel(
            encoder_decoder_type=encoder_decoder_type,
            encoder_decoder_name=encoder_decoder_name,
            args=model_args,
            model=self._backbone_model,
            use_cuda=True,
            cuda_device=cuda_device_no,
        )

    def predict(self, document: str) -> List[LabeledSegment]:
        with time_block("Took {:0.2f} seconds make seq2seq predictions"):
            results = self._model.predict([document])
            print(f"Output from seq2seq model: {results}")
        return [
            LabeledSegment(
                segment=pred,
                label="benefit",
                score=0.0
            )
            for pred in results
        ]
```